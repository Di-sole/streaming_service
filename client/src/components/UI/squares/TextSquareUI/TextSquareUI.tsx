import styles from './TextSquareUI.module.scss';

interface TextSquareUIProps {
    value: number;
    textSize?: 'medium' | 'small';
}

const TextSquareUI = ({ value, textSize = 'small' }: TextSquareUIProps) => {
    return (
        <div
            className={[
                styles.value,
                textSize === 'medium' ? styles.value_medium : '',
                value > 7.5 ? styles.value_high : '',
            ]
                .join(' ')
                .trim()}
            data-testid="textSquare"
        >
            {value}
        </div>
    );
};

export default TextSquareUI;
