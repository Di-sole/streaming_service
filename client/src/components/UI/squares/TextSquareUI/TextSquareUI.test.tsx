import { render, screen } from '@testing-library/react';
import TextSquareUI from './TextSquareUI';

describe('TEXTSQUAREUI SNAPSHOTS TESTS', () => {
    test('high small TextSquareUI should not be changed', () => {
        render(<TextSquareUI value={9.5} />);

        const div = screen.getByTestId('textSquare');
        expect(div).toMatchSnapshot();
    });

    test('no high medium TextSquareUI should not be changed', () => {
        render(<TextSquareUI value={6.5} />);

        const div = screen.getByTestId('textSquare');
        expect(div).toMatchSnapshot();
    });
});
