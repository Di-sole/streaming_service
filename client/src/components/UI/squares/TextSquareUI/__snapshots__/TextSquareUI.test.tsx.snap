// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEXTSQUAREUI SNAPSHOTS TESTS high small TextSquareUI should not be changed 1`] = `
<div
  class="value  value_high"
  data-testid="textSquare"
>
  9.5
</div>
`;

exports[`TEXTSQUAREUI SNAPSHOTS TESTS no high medium TextSquareUI should not be changed 1`] = `
<div
  class="value"
  data-testid="textSquare"
>
  6.5
</div>
`;
